pkg/middleware/logging/logging.go

```diff
@@ -49,13 +53,20 @@ func NewStreamingLoggingInterceptor(logger logger.Logger) grpc.StreamServerInter
 
 type reporter struct {
 	ctx            context.Context
+	grpcCtx        context.Context
+	start          time.Time
 	logger         logger.Logger
 	fields         []zap.Field
 	protomarshaler protojson.MarshalOptions
 }
 
 // PostCall is invoked after all PostMsgSend operations.
 func (r *reporter) PostCall(err error, _ time.Duration) {
+	duration := time.Since(r.start)
```

Comment:
I just realized that we don't need to record `r.start` to calculate the duration - the duration is the second parameter to this function 😉 https://github.com/grpc-ecosystem/go-grpc-middleware/blob/62b7de50cda5a5d633f1013bfbe50e0f38db34ef/interceptors/server.go#L21-L23